syncr_backend/util/drop_util.py

```diff
@@ -177,21 +181,24 @@ def get_file_metadata(
     """
     logger.info("getting file metadata for %s", crypto_util.b64encode(file_id))
     metadata_dir = os.path.join(save_dir, DEFAULT_FILE_METADATA_LOCATION)
-    metadata = FileMetadata.read_file(file_id, metadata_dir)
+    metadata = await FileMetadata.read_file(file_id, metadata_dir)
     if metadata is None:
         logger.debug("file metadata not on disk, getting from network")
-        metadata = send_requests.do_request(
+        metadata = await send_requests.do_request(
             request_fun=send_requests.send_file_metadata_request,
             peers=peers,
             fun_args={'drop_id': drop_id, 'file_id': file_id},
         )
 
-        metadata.write_file(metadata_dir)
+        if metadata is None:
+            raise Exception
```

Comment:
Add a specific Exception, probably should also have a logger.debug statement before it